### PR TITLE
[5.3] Merge a model into simple query

### PR DESCRIPTION
**Pull request by akiyamaSM**
A simple way to merge a model instance into a simple query

 $user = App\User::find(29);
 $users = \DB::table('users')->where('id', '>', 50)->union($user)->get()

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1516,12 +1516,33 @@ public function union($query, $all = false)
...
 
         $this->addBinding($query->getBindings(), 'union');
 
         return $this;
     }
+    
+    /**
+     * Merge model record into simple query.
+     *
+     * @param Model $model
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: please use fully qualified names and a leading slash in phpdoc

File: src/Illuminate/Database/Query/Builder.php [hunk 2]
```diff
@@ -1516,12 +1516,33 @@ public function union($query, $all = false)
...
+     * @return Builder|static
+     * @throws InvalidArgumentException
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: please format this to match the rest of our code

File: src/Illuminate/Database/Query/Builder.php [hunk 3]
```diff
@@ -1516,12 +1516,33 @@ public function union($query, $all = false)
...
+     */
+    protected function merge(Model $model)
+    {
+        if ($model->getTable() === $this->from) {
+            return $this->union(
+                \DB::table($this->from)->where($model->getKeyName(), $model->getKey())
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: Can't have a hack like that I'm afraid.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Database/Query/Builder.php (hunk 3 above)
Comment (akiyamaSM) [inline src/Illuminate/Database/Query/Builder.php]: I'll rewrite in an other way

File: src/Illuminate/Database/Query/Builder.php [hunk 4]
```diff
@@ -1524,6 +1529,24 @@ public function union($query, $all = false)
     }
 
     /**
+     * Merge model record into simple query.
+     *
+     * @param \Illuminate\Database\Eloquent\Model $model
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: 2 spaces before the slash and dollar please, not 1

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by GrahamCampbell** [comment]
Probably best to send this to 5.4.

**Comment by GrahamCampbell** [comment]
Yes, please send to 5.4, to avoid merge conflicts. :)

**Pull request closed without merging**